pkgcore/ebuild/processor.py

```diff
@@ -347,17 +347,18 @@ def __init__(self, userpriv, sandbox, fakeroot, save_file):
         # force to a neutral dir so that sandbox/fakeroot won't explode if
         # ran from a nonexistent dir
         spawn_opts["cwd"] = e_const.EAPI_BIN_PATH
-        # little trick. we force the pipes to be high up fd wise so
-        # nobody stupidly hits 'em.
-        max_fd = min(pkgcore.spawn.max_fd_limit, 1024)
+
+        # Use consecutive FDs because deriving from RLIMIT_NOFILE yields unexpected results on OSX.
+        read_fd = 3
```

Comment:
This has a pretty strong chance of intersecting with ebuilds that screw with FDs, or build processes- they should behave better, but it happens (hence the original code).

how exactly is rlimit_nofile giving unexpected results?  That's a bit of a standard posix thing afaik- osx *should* handle it correctly.  I'm reasonably sure portage is still using this same trick- which I know worked on osx way back in the prefix days.